pyvista/plotting/plotting.py

```diff
@@ -2414,6 +2416,10 @@ def add_mesh(
         else:
             self.mapper.SetScalarModeToUseFieldData()
 
+        # track if any data arrays have been added
+        if new_scalars:
```

Comment:
This is a bit fishy to me. When we look up scalars by name that are already present in the dataset, we get e.g. `new_scalars = (None, 'point')`. So we would add them to the `_added_scalars`. Of course there's no risk of these causing a problem, because we don't store the original name so we can't accidentally remove it from the dataset.

But shouldn't we do something like
```suggestion
        # track if any data arrays have been added
        if new_scalars is not None and new_scalars[0] is not None:
```
to filter out these cases?

Reply:
Implemented in 32fcae6f21b8213e6c988a2d78a960e9a0300d09.